osu.Game/Beatmaps/BeatmapStore.cs

```diff
@@ -125,46 +127,20 @@ public bool Restore(BeatmapInfo beatmap)
         private void cleanupPendingDeletions()
         {
             Connection.BeatmapSetInfo.RemoveRange(Connection.BeatmapSetInfo.Where(b => b.DeletePending && !b.Protected));
+            Connection.SaveChanges();
         }
 
-        public BeatmapSetInfo QueryBeatmapSet(Func<BeatmapSetInfo, bool> query)
-        {
-            return Connection.BeatmapSetInfo
-                .Include(b => b.Metadata)
-                .Include(b => b.Beatmaps).ThenInclude(b => b.Ruleset)
-                .Include(b => b.Beatmaps).ThenInclude(b => b.Difficulty)
-                .Include(b => b.Files).ThenInclude(f => f.FileInfo)
-                .FirstOrDefault(query);
-        }
-
-        public List<BeatmapSetInfo> QueryBeatmapSets(Expression<Func<BeatmapSetInfo, bool>> query)
-        {
-            return Connection.BeatmapSetInfo
-                .Include(b => b.Metadata)
-                .Include(b => b.Beatmaps).ThenInclude(b => b.Ruleset)
-                .Include(b => b.Beatmaps).ThenInclude(b => b.Difficulty)
-                .Include(b => b.Files).ThenInclude(f => f.FileInfo)
-                .Where(query).ToList();
-        }
-
-        public BeatmapInfo QueryBeatmap(Func<BeatmapInfo, bool> query)
-        {
-            return Connection.BeatmapInfo
-                .Include(b => b.BeatmapSet)
-                .Include(b => b.Metadata)
-                .Include(b => b.Ruleset)
-                .Include(b => b.Difficulty)
-                .FirstOrDefault(query);
-        }
-
-        public List<BeatmapInfo> QueryBeatmaps(Expression<Func<BeatmapInfo, bool>> query)
-        {
-            return Connection.BeatmapInfo
-                .Include(b => b.BeatmapSet)
-                .Include(b => b.Metadata)
-                .Include(b => b.Ruleset)
-                .Include(b => b.Difficulty)
-                .Where(query).ToList();
-        }
+        public IEnumerable<BeatmapSetInfo> BeatmapSets => Connection.BeatmapSetInfo

```

Comment:
I like these two properties.
Shame on me because I did not come to this implementation from the very beginning.